parse-nessus-xml.py

```diff
@@ -28,10 +29,88 @@
 print(f'File size: {file_size}')
 print(f'Scan start date: {start_date}')
 
+DAEMONS = """
+    bosh.dns
+    bosh.dns.adapter
+    bosh.dns.health
+    alertmanager
+    binding.cache
+    blackbox.exporter
+    bosh.exporter
+    cf.exporter
+    concourse
+    docker
+    doomsday
+    firehose.exporter
+    gdn
+    gnatsd
+    grafana.server
+    java
```

Comment:
`java` (and `ruby`) could be run as daemons for unknown processes, right? Do they belong here?

I think we should probably remove or refine them, depending on what the report shows. Either the report shows just the process without any params, in which case we don't know whether this is `java org.elasticsearch.Logstash` or `com.nefarious.Badstuff`. Or the report shows the process with params, in which case the anchored regex means we'll only ever match if we're running java as a daemon by itself

Maybe I'm overthinking

Reply:
Now you got me overthinking this, but I got to clean up some stuff and make this more rigorous in terms of name AND path.

Unfortunately, I python `re` doesn't support subpattern recursion or I could do something Perl-ish like:

```
\/var\/vcap\/data\/packates\/({DAEMONS})\/[a-z0-9]+\/(?1)
```
and ensure the binary matched the name of the directory, but that doesn't seem easy at this point (and not worth the effort)

Reply:
@pburkholder I have good or bad news, depending on your perspective.

Python's `re` does support backreferences, including named backreferences